### PR TITLE
v1.4.2 rc.1

### PR DESCRIPTION
**Pull request by dkarski**
v1.4.2 rc.1

**Description**

[CP-1444](https://appnroll.atlassian.net/browse/CP-1444) - fixed

### CODE REVIEW

File: packages/app/src/overview/components/overview-screens/pure-overview/pure-overview.component.test.tsx [hunk 1]
```diff
@@ -127,7 +130,40 @@ test("Renders Mudita pure data", () => {
...
-  expect(getByTestId(ErrorSyncModalTestIds.Container)).toBeInTheDocument()
+describe("`ErrorSyncModal` logic", () => {
+  test("when sync error occurred and `restoreDeviceState` is undefined modal is visible", () => {
+    const { queryByTestId } = render({
+      restoreDeviceState: undefined,
+      syncState: SynchronizationState.Error,
+    })
+    expect(queryByTestId(ErrorSyncModalTestIds.Container)).toBeInTheDocument()
+  })
+  test("when sync error occurred and `restoreDeviceState` has empty state is visible", () => {
```
Comment (wojcik92michal) [inline packages/app/src/overview/components/overview-screens/pure-overview/pure-overview.component.test.tsx:141]:
```suggestion
  test("when sync error occurred and `restoreDeviceState` has empty state modal is visible", () => {
```

File: packages/app/src/overview/components/overview-screens/pure-overview/pure-overview.component.test.tsx [hunk 2]
```diff
@@ -127,7 +130,40 @@ test("Renders Mudita pure data", () => {
...
+    const { queryByTestId } = render({
+      restoreDeviceState: RestoreDeviceDataState.Empty,
+      syncState: SynchronizationState.Error,
+    })
+    expect(queryByTestId(ErrorSyncModalTestIds.Container)).toBeInTheDocument()
+  })
+  test("when sync error occurred and `restoreDeviceState` has error state isn't visible", () => {
```
Comment (wojcik92michal) [inline packages/app/src/overview/components/overview-screens/pure-overview/pure-overview.component.test.tsx:148]:
```suggestion
  test("when sync error occurred and `restoreDeviceState` has error state modal isn't visible", () => {
```

File: packages/app/src/overview/components/overview-screens/pure-overview/pure-overview.component.test.tsx [hunk 3]
```diff
@@ -127,7 +130,40 @@ test("Renders Mudita pure data", () => {
...
+    const { queryByTestId } = render({
+      restoreDeviceState: RestoreDeviceDataState.Error,
+      syncState: SynchronizationState.Error,
+    })
+    expect(queryByTestId(ErrorSyncModalTestIds.Container)).not.toBeInTheDocument()
+  })
+  test("when sync error occurred and `restoreDeviceState` has running state isn't visible", () => {
```
Comment (wojcik92michal) [inline packages/app/src/overview/components/overview-screens/pure-overview/pure-overview.component.test.tsx:155]:
```suggestion
  test("when sync error occurred and `restoreDeviceState` has running state modal isn't visible", () => {
```

File: packages/app/src/overview/components/overview-screens/pure-overview/pure-overview.component.test.tsx [hunk 4]
```diff
@@ -127,7 +130,40 @@ test("Renders Mudita pure data", () => {
...
+    const { queryByTestId } = render({
+      restoreDeviceState: RestoreDeviceDataState.Running,
+      syncState: SynchronizationState.Error,
+    })
+    expect(queryByTestId(ErrorSyncModalTestIds.Container)).not.toBeInTheDocument()
+  })
+  test("when sync error occurred and `restoreDeviceState` has finished state isn't visible", () => {
```
Comment (wojcik92michal) [inline packages/app/src/overview/components/overview-screens/pure-overview/pure-overview.component.test.tsx:162]:
```suggestion
  test("when sync error occurred and `restoreDeviceState` has finished state modal isn't visible", () => {
```

**Review by wojcik92michal — APPROVED** [review]

File: packages/app/src/overview/components/overview-screens/pure-overview/pure-overview.component.test.tsx (hunk 4 above)
Comment (dkarski) [reply packages/app/src/overview/components/overview-screens/pure-overview/pure-overview.component.test.tsx]: https://github.com/mudita/mudita-center/pull/846/commits/c45ef903324aa73c0aa9314a50b45b475caf80bf

File: packages/app/src/overview/components/overview-screens/pure-overview/pure-overview.component.test.tsx (hunk 3 above)
Comment (dkarski) [reply packages/app/src/overview/components/overview-screens/pure-overview/pure-overview.component.test.tsx]: https://github.com/mudita/mudita-center/pull/846/commits/c45ef903324aa73c0aa9314a50b45b475caf80bf

**Review by igorbezsmertnyi — APPROVED** [review]

**Pull request merged**